### DISCUSSION TITLE
RaspberryPi hardware and/or software video encoding

### DISCUSSION THREAD

**Discussion by beyonlo** [bluenviron/mediamtx#3916] [Questions]
### Question

Hi all.

I already asked this question in another discussion (#3914), but my question was completely off topic with that subject, so now I'm asking with the correct subject/title and better content.

Since the pi5 does not have the hardware video encoding feature like the pi4, pi3 and pi2w, I would like to ask: does mediamtx use automatically hardware video encoding when used in pi4, pi3 and pi2w and software video encoding in the pi5, or is needed to do an extra config (how?) depending what RaspberryPi hardware version will be used?

Thank you!

**Comment by aler9** [comment]
Hello,

> does mediamtx use automatically hardware video encoding when used in pi4, pi3 and pi2w and software video encoding in the pi5

absolutely yes, the behavior is exactly the one you described.

> or is needed to do an extra config (how?) depending what RaspberryPi hardware version will be used?

you can override the default behavior by using setting `rpiCameraCodec`:

```yml
paths:
  rpi:
    source: rpiCamera
    # available values: automatic, hardwareH264, softwareH264 
    rpiCameraCodec: hardwareH264
```

Please refer to the [configuration file](https://github.com/bluenviron/mediamtx/blob/main/mediamtx.yml) for all available (and up-to-date) settings.
